### DISCUSSION TITLE
LineBot for Disfactory

### DISCUSSION THREAD

**Discussion by DarkenSiva** [Disfactory/about.disfactory.tw#56] [Show and tell]
目前有個由 GeoThings Inc. 提供 (在這次的第一百次聚會的同時所製作) 的 LineBot 可以當成回報的另一個介面。以下摘錄 ael 在 Slack 的筆記如下：

1. 使用 LINE 的顧慮：會收到使用者 ID 和頭像，無法維持在鄉間實地回報者的匿名性（擔心地方勢力糾葛）目前網站設計是聯絡資訊選填，只有不到一半的回報者會填稱呼和聯絡資訊。但是 LINE 真的是容易讓目標使用者上手的互動平台。
如果可以設計成讓使用者清楚知道，使用 LINE 回報會紀錄什麼，或是拿來做比較不敏感的行為，例如辨識已有的資料（類似大家來找廠），也會很棒。
2. 或是使用 Telegram ，但是一般台灣使用者不熟悉。
3. 究心這個 LineBot 版本是不限使用者人數，並且可以直接傳送照片、影片、以及地點資訊來完成回報。跟之前的 prototype 功能不同。

以下是 Disfactory 非官方 LineBot 的測試 QRCode，使用 Line 掃描並加入好友即可開始進行回報。流程就是 拆 -> 拍照 -> 打卡 三步驟結束。目前還沒串接 Disfactory 的後臺，這算是 pre-alpha 內部測試版本吧。總之歡迎各位給我們建議，並可以整合討論 隱私、易用、普及程度 等面向的議題。謝謝!

![Disfactory](https://user-images.githubusercontent.com/6604052/139419611-300688d6-368a-48c9-8ba1-aaccac74cbc1.png)

**Comment by aelcenganda** [comment]
Oriyar 和 slayer 聊得滿開心的，想要探索做這個 LINE bot 對倡議帶來的可能性 🥰

昨天小聚有跟幾個人聊了一下，大家都樂觀其成（非常開心不用自己寫？）。我這邊從社群貢獻者和產品經理的角度給一些需要思考的問題：

**合作：**
1. 合作互惠的方式（主要是基金會＋公司）
2. 專案目標？（時程、規模、期待結果）
3. 開發能量（目前 Disfactory 社群這邊，至少到過年前社群人力都吃緊，只能做低度維運。前後端幾乎都沒空，設計師可能比較有空）


**產品策略：**
1. 如何推廣 LINE bot？
    - 之前使用者研究發現的一些事：
        - 對 2020 年第一批有留聯絡資訊的實地回報使用者研究，當初是希望找到在地 LINE 群組可以擴散，結果發現訪問的五六個人中沒有任何人想要分享 Disfactory 到社群媒體上，因為他們完全不想讓身邊的人知道他們回報工廠（地方很小，都是認識的人，不想起衝突）。扣掉環保團體組織的回報行動，個人使用者多半是從新聞和 Google 搜尋上進來的。
        - 很多回報的使用者是回家用電腦上傳，因為怕定位錯誤以及不想在現場停留太久，所以我們一開始預設的使用者用手機拍照，直接在工廠附近上傳的這個情境，不常見。手機會拿來拍照和瀏覽，但是上傳地點大部分會回到電腦上。
        - 最終實際回報的使用者，都是在地人，不熟悉當地還有農舍、倉庫、工廠差別的人，很難揪團或路過回報。而且有不少使用者是先走檢舉污染，環保局那邊走不通再來用農地這個不符合土地利用的法規去檢舉。通常只會檢舉家裏附近有污染、噪音影響的，除了在地環保團體外，不太會有高頻或重度多次回報使用者。
        - 使用者非常想要知道回報之後，該工廠的公文、政府回應處理進度，但目前的 disfactory 網站介面，這個功能不好用，需要存取該工廠網址（用經緯度做 unique url），或是使用者自行拖拉地圖到檢舉地圖去確認公文進度。這是我覺得 LINE bot 最有優勢的地方，可以精準更新 follow up status to users。
    - 所以這次 LINE bot 要如何推廣？也許 LINE bot 搭配社區教育推廣，會比現在的網站介面好用滿多的，可以開發新客群。或是從 GeoThing 那邊學習如何推廣 LINE bot 和設計使用情境。目前有訪問到回報者年紀多半是 30-50，年齡層更高的偏少。
2. 客製化的程度要多高？問答流程設計、實地使用者測試，公司、基金會、社群可以怎麼合作？
3. Geo Thing 的產品看起來會搭配後台，這個後台的資料欄位、介面跟 disfactory.tw 既有架在 Django 的後台案件管理，要如何串接或是管理？
4. GeoThing 有想要藉這個機會，測試某些功能或是嘗試新東西嗎？（除了 CSR、SDGs 這些企業品牌形象以外，產品面有沒有可以回饋 GeoThing 的地方？）
5. 社群專案除了加上 LINE bot 這個回報介面增加回報人數之外，這次開發可以如何回饋社群和基金會？

@deeper747 你要幫補目前每月進站人次、回報人次、回報案件的數據嗎？


**技術：**
1. 當初用網站套疊地圖圖層時，有農地/非農地圖層，鎖住非農地，不讓使用者回報。 LINE bot 看要同樣加上過濾，還是所有點位都收？要過濾土地用途的話，技術上要怎麼處理？
2. 社群需要多開哪些 API？（已有的 [openAPI](https://api.disfactory.tw/redoc/)）
3. 資料會如何處理？
    - 會收集到哪些個資？不想收集的話，如何避開、不紀錄、或是銷毀？目前 disfactory.tw 不需要登入、不記 user ip，聯絡資料為選填。上傳的照片在公開平台已移除 EXIF，也就是相片拍攝時的時間、地點、裝置的資訊都移除了。正在開發中的衛星空照圖比對 SpotDiff 也是不拿 IP、沒有帳號登入，只拿 GA client_id
    - 資料授權條款： LINE 跟使用者有一個使用條款， LINE@ 帳號跟使用者會有另外一個（跟究心科技簽的），所以從 GeoThing LINE bot 收到的回報資料，智財權在誰那裡？會用什麼授權開放？目前 disfactory.tw 的[貢獻授權協議](https://license.disfactory.tw/)

**Comment by deeper747** [comment]
> @deeper747 你要幫補目前每月進站人次、回報人次、回報案件的數據嗎？

每月
- 進站人次：877
- 更新次數：137
- 有更新、新增工廠數：63

**Comment by Yukaii** [comment]
想到也可以用 [LIFF](https://developers.line.biz/en/docs/liff/overview/) 的方式內嵌一個網頁版 disfactory.tw 進去，並在成功新增工廠後用 [`liff.sendMessages()`](https://developers.line.biz/en/reference/liff/#send-messages) 把工廠的連結傳給使用者
